scripts/hoist_binstall_archives.py

```
@@ -0,0 +1,378 @@
+    """
+    completed: list[tuple[Path, Path]] = []
+    try:
+        for entry in located:
+            for source in (entry.archive, entry.sidecar):
+                destination = dist_dir / source.name
+                shutil.move(source, destination)
+                completed.append((source, destination))
+    except BaseException as failure:
+        try:
+            _rollback_completed_moves(completed)
+        except BaseException as rollback_failure:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `BaseException`

(BLE001)